### PR TITLE
Kafka 2.6.0 client upgrade and drop Scala 2.11 support

### LINKED ISSUE #1098: Upgrade to Kafka 2.6.0 client
Now that [Kafka 2.5.0 is released](https://kafka.apache.org/downloads#2.5.0) Alpakka Kafka should check if we there is anything in particular to adapt to.

Kafka dropped Scala 2.11 which doesn't affect the client library, but the Embedded Kafka will not be able to work with Scala 2.11 anymore.

See https://github.com/akka/alpakka-kafka/issues/1012

### PR DESCRIPTION
**Pull request by seglo**
Fixes #1098

Upgrade to Apache Kafka 2.6.0.

* Kafka 2.5.0 drops support for Scala 2.11.  We'll stop publishing 2.11 builds of Alpakka Kafka as well to avoid the conflict of Scala 2.11 issues using Embedded Kafka with the testkit. #1012.
* Drop remaining embedded kafka tests, except those required for documentation.

### CODE REVIEW

**Comment by ennru** [comment]
Let's update this for Kafka 2.6.0.

**Comment by seglo** [comment]
There are two remaining embedded-kafka tests that are consistently failing for some reason in the 2.13 build.  The broker of is logging the following exception:

```
[info] 08:54:28.624 ERROR [ka-request-handler-6] kafka.server.KafkaApis                [KafkaApi-0] Error when handling request: clientId=adminclient-1, correlationId=101, api=METADATA, version=9, body={topics=[],allow_auto_topic_creation=true,include_cluster_authorized_operations=false,include_topic_authorized_operations=false,_tagged_fields={}}
[info] java.lang.NoSuchMethodError: 'org.apache.kafka.common.requests.MetadataResponse org.apache.kafka.common.requests.MetadataResponse.prepareResponse(int, java.util.Collection, java.lang.String, int, java.util.List, int)'
[info]  at kafka.server.KafkaApis.handleTopicMetadataRequest(KafkaApis.scala:1177)
[info]  at kafka.server.KafkaApis.handle(KafkaApis.scala:131)
[info]  at kafka.server.KafkaRequestHandler.run(KafkaRequestHandler.scala:70)
[info]  at java.base/java.lang.Thread.run(Thread.java:834)
```

It's possible this is due to some version mismatch of embedded-kafka and Kafka client libraries, but it's unclear why it's only an issue with Scala 2.13.  At the time of this comment there is not 2.6.0 version of embedded-kafka.

Options:

1) Wait until embedded-kafka 2.6.0 is released
2) Build our own embedded-kafka 2.6.0
3) Don't run the remaining embedded-kafka tests (at least for Scala 2.13) since we're going to replace them before the next release anyway.

File: project/project-info.conf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 project-info {
   version: "current"
-  scala-versions: ["2.11", "2.12"]
+  scala-versions: ["2.12", "2.13"]
```
Comment (ennru) [inline project/project-info.conf:3]: I believe we can leave this out altogether, it should be picked up from cross scala versions.

**Comment by seglo** [comment]
embedded-kafka 2.6.0 has been released.

File: docs/src/main/paradox/testing.md [hunk 2]
```diff
@@ -89,7 +89,7 @@ Add the Embedded Kafka to your test dependencies:
 
 As Kafka uses Scala internally, only the Scala versions supported by Kafka can be used together with Embedded Kafka. To be independent of Kafka's supported Scala versions, run @ref:[Kafka in a Docker container](testing-testcontainers.md).
 
-The helpers for running Embedded Kafka are available for **Scala 2.11 and 2.12**.
+The helpers for running Embedded Kafka are available for **Scala 2.12**.
```
Comment (ennru) [inline docs/src/main/paradox/testing.md:92]: And Scala 2.13?

File: project/project-info.conf (hunk 1 above)
Comment (seglo) [reply project/project-info.conf:3]: Leave out the whole `scala-versions` property?

File: docs/src/main/paradox/testing.md (hunk 2 above)
Comment (seglo) [reply docs/src/main/paradox/testing.md:92]: I'll update it, but we're going to rearrange this whole section when (if?) we drop embedded-kafka.

File: project/project-info.conf (hunk 1 above)
Comment (seglo) [reply project/project-info.conf]: Seems to be when it's dropped :+1:

**Comment by ennru** [comment]
Please rebase and create an initial version of `release-notes/2.1.x.md`.

**Review by ennru — APPROVED** [review]
LGTM.

**Pull request merged**